### PR TITLE
Implement Docker Compose

### PR DESCRIPTION
**Pull request by dleitee**
Related to #8 

@CheesecakeLabs/backend

### CODE REVIEW

**Review by smaniotto — APPROVED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM python:3.5
+ENV PYTHONUNBUFFERED 1
+RUN mkdir /code
+WORKDIR /code
+ADD requirements.txt /code/
+RUN pip install -r requirements.txt
```
Comment (joaoricardo000) [inline Dockerfile]: We could use `python:3.5-onbuild` base image:  
https://github.com/docker-library/python/blob/master/3.5/onbuild/Dockerfile  
```
FROM python:3.5

RUN mkdir -p /usr/src/app
WORKDIR /usr/src/app

ONBUILD COPY requirements.txt /usr/src/app/
ONBUILD RUN pip install --no-cache-dir -r requirements.txt

ONBUILD COPY . /usr/src/app
```  

Note the pip install before the code copy, it reduces the rebuild time.
I also find useful let the default CMD on the Dockerfile, for reference and commodity.

Comment (dleitee) [reply Dockerfile]: 👍

Comment (jonatasbaldin) [reply Dockerfile]: :awesome_face:

Comment (jonatasbaldin) [reply Dockerfile]: Actually, we need to test the `COPY` and `volume` together, so we can have a deploy and development environment in the same place.

Comment (jonatasbaldin) [reply Dockerfile]: Hey @joaoricardo000, you mean letting the CMD (`python manage.py runserver` etc) on the Dockerfile?

**Comment by jonatasbaldin** [comment]
Sup everybody, we added some configuration to use Compose just in development environment, separating it from the Dockerfile. What do u think about that? @CheesecakeLabs/backend

**Comment by joaoricardo000** [comment]
About the development/production difference, it is good to have separeted enviroment, but for production we would not use the `manage.py runserver`, probably a `gunicorn`.  
That would require a new dependency on `requirements.txt`, not only passing a command via `CMD`.  
But that's the only thing I can think about now, the rest is perfect!

**Comment by jonatasbaldin** [comment]
Yeah, totally! Production will use another WSGI, and we need to discuss and configure that as well. Will open an issue.

**Pull request merged**